controller/linodemachine_controller.go

```diff
@@ -125,25 +118,38 @@ func (r *LinodeMachineReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 		return ctrl.Result{}, err
 	}
 
-	machine, err := getOwnerMachine(ctx, r.TracedClient(), *linodeMachine, log)
+	machine, err := kutil.GetOwnerMachine(ctx, r.TracedClient(), linodeMachine.ObjectMeta)
 	if err != nil || machine == nil {
 		return ctrl.Result{}, err
 	}
 	log = log.WithValues("LinodeMachine", machine.Name)
 
-	cluster, err := r.getClusterFromMetadata(ctx, *machine, log)
+	cluster, err := kutil.GetClusterFromMetadata(ctx, r.TracedClient(), machine.ObjectMeta)
 	if err != nil || cluster == nil {
 		return ctrl.Result{}, err
 	}
 
+	linodeClusterKey := client.ObjectKey{
+		Namespace: linodeMachine.Namespace,
+		Name:      cluster.Spec.InfrastructureRef.Name,
+	}
+	linodeCluster := &infrav1alpha2.LinodeCluster{}
```

Comment:
I like this approach. But now that we are populating the field here, we should remove the `Get()` calls in the reconcile method